src/core/statistics.cpp

```diff
@@ -51,7 +51,7 @@
 
 /** Previous particle configurations (needed for offline analysis and
     correlation analysis) */
-double **configs = nullptr;
+std::vector<std::vector<double>> configs;
```

Comment:
This should be a `std::vector<Utils::Vector3d>`, the inner size is known statically. This is really wasteful...

Reply:
If the number of particles changes in a simulation then the inner size is not statically known but changes during the runtime: `  configs[n_configs].resize(3 * n_part_conf);` (in analyze_configs() in statistics.cpp)

One could introdce a `std::vector<std::vector<Utils::Vector3d>> configs`,
which one would enlarge via configs[n_configs].resize(n_part_conf) and adress like `configs[n_configs][particle_id]=p.r.p`, however I am not sure what the benefit is compared to:
`  int i = 0;
  for (auto const &p : partCfg) {
    configs[n_configs][3 * i + 0] = p.r.p[0];
    configs[n_configs][3 * i + 1] = p.r.p[1];
    configs[n_configs][3 * i + 2] = p.r.p[2];
    i++;
  }`

Reply:
The inner dimension is, as you are saying, `3`. The advantage would be that e.g. it's not possible to have an incompatible size, also the code get less noise, because there is no need for elementwise copy.

Reply:
How about saving this for an independent PR. The corresponding changes would require further changes in the way configs[I] is handled? This is not related to memory management though.